sql/catalyst/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -508,4 +516,38 @@ final ColumnarMap getMap(int rowId) {
       super(vector);
     }
   }
+
+  private static class IntervalYearAccessor extends ArrowVectorAccessor {
+
+    private final IntervalYearVector accessor;
+
+    IntervalYearAccessor(IntervalYearVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    int getInt(int rowId) {
+      return accessor.get(rowId);
+    }
+  }
+
+  private static class IntervalDayAccessor extends ArrowVectorAccessor {
+
+    private final IntervalDayVector accessor;
+    private final NullableIntervalDayHolder intervalDayHolder = new NullableIntervalDayHolder();
+
+    IntervalDayAccessor(IntervalDayVector vector) {
+      super(vector);
+      this.accessor = vector;
+    }
+
+    @Override
+    long getLong(int rowId) {
+      accessor.get(rowId, intervalDayHolder);
+      return Math.addExact(
+              Math.multiplyExact(intervalDayHolder.days, MICROS_PER_DAY),
+              Math.multiplyExact(intervalDayHolder.milliseconds, MICROS_PER_MILLIS));
```

Comment:
@MaxGekk  `intervalDayHolder.days` and `intervalDayHolder.milliseconds` are int, can they really overflow?

Reply:
I think so:
```scala
scala> Long.MaxValue / MICROS_PER_DAY
res0: Long = 106751991

scala> (106751991 + 1) * MICROS_PER_DAY
res1: Long = -9223371964909551616
```

Reply:
but Int.Max * MICROS_PER_MILLIS won't overflow, right?

Reply:
Right

Reply:
I use "intervalDayHolder.days * MICROS_PER_DAY" instead of Math.multiplyExact

Reply:
@Peng-Lei  Wenchen asked about milliseconds part, why did you changed the days multiplication? I showed above that it can overflow Long.

Reply:
Sorry，My mistake. I get it. I'll fix it right now.